### PR TITLE
Merge release to main

### PR DESCRIPTION
**Pull request by erick-xanadu**
Merging release changes back to main

### CODE REVIEW

File: .dep-versions [hunk 1]
```diff
@@ -5,5 +5,5 @@ llvm=cd9a641613eddf25d4b25eaa96b2c393d401d42c
 enzyme=1beb98b51442d50652eaa3ffb9574f4720d611f1
 
 # Always remove custom PL/LQ versions before release.
-pennylane=f638bc53a09724a83d9d58964bf37bfd438e6aa3
-lightning=0.35.0-dev21
+pennylane=f9bdb32778b7384c43bb5bb17254b8a3d054e974 # Latest rc-sync to master
+lightning=0.35.0
```
Comment (dime10) [inline .dep-versions:9]: I think these should go now until we dependent on development functionality again right?
```suggestion
```

**Review by dime10 — COMMENTED** [review]
I noticed the pytest pin in the wheels is no longer present in the merge, is that intended?

**Comment by dime10** [comment]
The lightning tag also needs to be updated everywhere 0.35 I think (makefile & wheel actions).

**Comment by erick-xanadu** [comment]
> I noticed the pytest pin in the wheels is no longer present in the merge, is that intended?

@dime10, yes. That version was yanked out of pypi. https://pypi.org/project/pytest/#history

**Comment by dime10** [comment]
> > I noticed the pytest pin in the wheels is no longer present in the merge, is that intended?
> 
> @dime10, yes. That version was yanked out of pypi. https://pypi.org/project/pytest/#history

But the pin is present in the requirements.txt

**Comment by erick-xanadu** [comment]
@dime10 I'll remove it from requirements.txt! Thanks for pointing it out!

File: .github/workflows/check-pl-compat.yaml [hunk 2]
```diff
@@ -111,7 +111,7 @@ jobs:
         CXX_COMPILER=$(which g++ }}) \
         ENABLE_LLD=OFF \
         RT_BUILD_DIR="$(pwd)/runtime-build" \
-        LIGHTNING_GIT_TAG_VALUE=master \
+        LIGHTNING_GIT_TAG_VALUE=0.35.0 \
```
Comment (dime10) [inline .github/workflows/check-pl-compat.yaml:114]: Oh no not these sorry, they always stay the same
```suggestion
        LIGHTNING_GIT_TAG_VALUE=master \
```

File: .github/workflows/check-pl-compat.yaml [hunk 3]
```diff
@@ -122,7 +122,7 @@ jobs:
         C_COMPILER=$(which gcc }}) \
         CXX_COMPILER=$(which g++ }}) \
         RT_BUILD_DIR="$(pwd)/runtime-build" \
-        LIGHTNING_GIT_TAG_VALUE=latest_release \
+        LIGHTNING_GIT_TAG_VALUE=0.35.0 \
```
Comment (dime10) [inline .github/workflows/check-pl-compat.yaml:125]:
```suggestion
        LIGHTNING_GIT_TAG_VALUE=latest_release \
```

**Review by dime10 — APPROVED** [review]
Looks good 💯

**Comment by dime10** [comment]
@erick-xanadu looks like the correct tag is `v0.35.0`

**Pull request merged**